src/resources/accounts.js

```diff
@@ -82,63 +95,79 @@ class Accounts {
     }
   }
 
-  async createDepositAddress(currency) {
-    if (!this.id) {
-      throw new SDKError(401, '[Zabo] Account not yet connected. See: https://zabo.com/docs#connecting-a-user')
-    } else if (!currency || typeof currency !== 'string') {
+  async createDepositAddress({ accountId = this.id, currency }) {
+    if (!accountId) {
+      if (utils.isNode()) {
+        throw new SDKError(400, '[Zabo] Missing `accountId` parameter. See: https://zabo.com/docs#create-a-deposit-address')
+      } else {
+        throw new SDKError(401, '[Zabo] Account not yet connected. See: https://zabo.com/docs#connecting-a-user')
+      }
+    }
+
+    if (!currency || typeof currency !== 'string') {
       throw new SDKError(400, '[Zabo] Missing or invalid `currency` parameter. See: https://zabo.com/docs##create-a-deposit-address')
     }
 
-    const providersWithStaticDepositAddresses = [
-      'metamask',
-      'ledger',
-      'hedera',
-      'address-only',
-      'binance',
-    ]
-
-    for (let provider of providersWithStaticDepositAddresses) {
-      if (provider === this.data.wallet_provider.name) {
-        console.warn(`[Zabo] Provider '${provider}' does not support dynamic address generation. Fallbacking to accounts.getDepositAddress()... More details: https://zabo.com/docs#get-deposit-address`)
-        return this.getDepositAddresses(currency)
+    if (utils.isBrowser()) {
+      const providersWithStaticDepositAddresses = [
+        'metamask',
+        'ledger',
+        'hedera',
+        'address-only',
+        'binance',
+      ]
+
+      for (let provider of providersWithStaticDepositAddresses) {
+        if (provider === this.data.wallet_provider.name) {
+          console.warn(`[Zabo] Provider '${provider}' does not support dynamic address generation. Fallbacking to accounts.getDepositAddress()... More details: https://zabo.com/docs#get-deposit-address`)
+          return this.getDepositAddresses(currency)
+        }
       }
     }
 
     try {
-      return this.api.request('POST', `/accounts/${this.id}/deposit-addresses?currency=${currency}`)
+      return this.api.request('POST', `/accounts/${accountId}/deposit-addresses?currency=${currency}`)
```

Comment:
In this request, is this using HMAC?

Reply:
Yes. Actually, the auth method is defined in the api based on the env (node vs browser).
Here is the conditional... https://github.com/zabo-api/zabo-sdk-js/blob/feature/ZAPI-1157-accounts-resource-server-side/src/api.js#L136